src/components/Auth/AuthCheckProcessor.ts

```diff
@@ -0,0 +1,173 @@
+import { EventEmitter2 } from 'eventemitter2'
+import { Config } from './AuthConfig'
+import { FaceTecSDK } from '~auth-sdk/FaceTecSDK.js/FaceTecSDK'
+import type {
+  FaceTecSessionResult,
+  FaceTecFaceScanResultCallback,
+  FaceTecFaceScanProcessor,
+} from '~auth-sdk/FaceTecSDK.js/FaceTecPublicApi'
+
+export class AuthCheckProcessor implements FaceTecFaceScanProcessor {
+  latestNetworkRequest = new XMLHttpRequest()
+  success
+  sdkToken
+  nextStep
+  events
+
+  constructor(
+    sessionToken: string,
+    sdkToken: string | undefined,
+    nextStep: () => void,
+    events?: EventEmitter2.emitter
+  ) {
+    this.sdkToken = sdkToken
+    this.success = false
+    this.nextStep = nextStep
+    this.events = events
+
+    //
+    // Part 1:  Starting the FaceTec Session
+    //
+    new FaceTecSDK.FaceTecSession(this, sessionToken)
+  }
+
+  //
+  // Part 2:  Handling the Result of a FaceScan
+  //
+  processSessionResultWhileFaceTecSDKWaits = (
+    sessionResult: FaceTecSessionResult,
+    faceScanResultCallback: FaceTecFaceScanResultCallback
+  ): void => {
+    //
+    // Part 3:  Handles early exit scenarios where there is no FaceScan to handle -- i.e. User Cancellation, Timeouts, etc.
+    //
+    if (
+      sessionResult.status !==
+      FaceTecSDK.FaceTecSessionStatus.SessionCompletedSuccessfully
+    ) {
+      console.info(
+        `Session was not completed successfully, cancelling.  Status: ${sessionResult.status}`
+      )
+      this.latestNetworkRequest.abort()
+      faceScanResultCallback.cancel()
+      return
+    }
+    //
+    // Part 4:  Get essential data off the FaceTecSessionResult
+    //
+    const parameters = {
+      face_scan: sessionResult.faceScan,
+      audit_trail_image: sessionResult.auditTrail[0],
+      low_quality_audit_trail_image: sessionResult.lowQualityAuditTrail[0],
+      metadata: {
+        sdk_source: 'onfido_web_sdk',
+        sdk_version: '1.0',
+        sdk_metadata: {
+          system: {
+            fingerprint: '',
+            model: '',
+            manufacturer: '',
+            brand: '',
+            product: '',
+            hardware: '',
+          },
+        },
+      },
+    }
+
+    //
+    // Part 5:  Make the Networking Call to Your Servers.  Below is just example code, you are free to customize based on how your own API works.
+    //
```

Comment:
As mentioned here to kimmi: https://github.com/onfido/onfido-sdk-ui/pull/1414#discussion_r629281729

Reply:
I'll re-edit the comments to better address whoever would benefit from these comments. Or would you rather I remove altogether?

Reply:
I think it makes sense to leave the comments when they are reflecting what the code does. So here I would only have 
`Part 5:  Make the Networking Call to the Onfido Servers.` 

> Below is just example code

This should be removed because the code is not an example and if I change anything in that code it would break the integration (unless I know what I'm doing) :)